### ISSUE TITLE
Syntax error points to wrong position in json5

### ISSUE THREAD

**Issue by plinnie** [dpranke/pyjson5#47]
When there is an error in a json5 file, the error message seems to always point to the first element in the json5 file. 

For example:
```python
import json5
JSON5 = """
{  
    version: "1.0",
    author: "John Smith",
    people : [
        "Monty",
        "Python"foo
    ]    
}
"""
json5.loads(JSON5)
```

Gives the error:

```
Traceback (most recent call last):
  File "C:/Users/Admin/surfdrive/client/stuff/tests/testjson5.py", line 12, in <module>
    json5.loads(JSON5)
  File "C:\Users\Admin\Anaconda3\envs\python38\lib\site-packages\json5\lib.py", line 82, in loads
    raise ValueError(err)
ValueError: <string>:3 Unexpected "v" at column 5
```
I would expect the error: `ValueError: <string>:7 Unexpected "f" at column 16.`

json5 version: 0.9.6
OS: Windows 10
Python: 3.8.8

**Comment by dpranke** [comment]
Looks like this was introduced with the fix for issue #26, which was (partially) attempting to fix a different case where the position was misreported. I'll have to figure out some way to handle both cases ...

**Comment by thepackrat** [comment]
+1 on this one, just cost me some sanity-questioning until I figured out what the parser error was doing.

**Comment by dpranke** [comment]
This is fixed in beb97a9d. I've just pushed v0.9.8, which contains this fix, to PyPI. 

Sorry this took so long for a fix, there were $REASONS why I wasn't on top of this, plus once I started looking at it, the bug ended up being a rather subtle problem in the generated parsing code (i.e., in the parser generator).
